src/formatDate.js

```diff
@@ -20,6 +11,29 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateWithoutSeparator = date.replace(/\D/g, ' ');
+
+  const separator = toFormat[3];
+
+  const formattedDate = dateWithoutSeparator.split(' ');
+
+  if (toFormat[2].length === 2) {
+    for (let i = 0; i < formattedDate.length; i++) {
+      if (formattedDate[i].length > toFormat[0].length) {
+        formattedDate[i] = formattedDate[i].split('').splice(-2, 2).join('');
+      }
+    }
+  }
+
+  if (fromFormat[0] !== toFormat[0]) {
+    [formattedDate[0], formattedDate[1]] = [formattedDate[1], formattedDate[0]];
+  }
+
+  if (fromFormat[0].includes('YY')) {
+    return formattedDate.reverse().join(`${separator}`);
+  }
+
+  return formattedDate.join(`${separator}`);
```

Comment:
I cant surely say that it is bad code. May be it happends beacuse of I am too silly. But I cant understand your code, I cant understand what is happening here.